### DISCUSSION TITLE
STT doesn't turn on

### DISCUSSION THREAD

**Discussion by TonySimonovsky** [danny-avila/LibreChat#3175] [Troubleshooting]
### What happened?

I added this to librechat.yaml as per https://www.librechat.ai/docs/configuration/stt_tts:
`stt:
  openai:
    apiKey: '${STT_API_KEY}'
    model: 'whisper-1'`

And added STT_API_KEY value (open api key) to .env.

STT still doesn't see to pick up as the transcription is much worse than normally with whisper so I conclude LibreChat is still using iPhone's tranacription.

2 additional things:
- in Chrome iOS in many cases no transcription happening at all
- in Safari MacOS I get "Microphone permission not granted" when trying to record audio though the permission is actually granted (didn't change after restarting Safari).

### Steps to Reproduce

...

### What browsers are you seeing the problem on?

Chrome, Mobile (iOS)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can only test on iOS but seems to be working fine.

<img width="414" alt="image" src="https://github.com/danny-avila/LibreChat/assets/110412045/9663b255-38c3-4d16-badc-b2ca9a70ed4a">

Are there any relevant logs in the server or browser console? You can check ./logs in project directory (if using docker), or check terminal of running server

The permission granted error could be due to a variety of browser incompatibilities. For example, chrome/safari work for me on iOS but not Firefox. Firefox works for me on desktop.

It’s tricky to balance all the browser APIs required but I think we can clean it up further @berry-13 

also adding debug logs to confirm STT requests are being made. In general, if you see the spinner, the audio data was sent and a response is incoming

**Reply by TonySimonovsky** [reply]
Thanks for the quick answer, Danny!

Browser console logs. I see nothing (using this method https://devtoolstips.org/tips/en/view-logs-for-other-browsers-on-ios/).

In the docker error logs here's what I see:
`
{"code":"ENOENT","errno":-2,"level":"error","message":"Config file YAML format is invalid: ENOENT: no such file >
{"level":"error","message":"Failed to get voices: Configuration or TTS schema is missing"}
{"level":"error","message":"Failed to get voices: Configuration or TTS schema is missing"}
{"level":"error","message":"Failed to get voices: Configuration or TTS schema is missing"}
{"level":"error","message":"Failed to get voices: Configuration or TTS schema is missing"}
{"code":"ENOENT","errno":-2,"level":"error","message":"Config file YAML format is invalid: ENOENT: no such file >
`

The "Config file YAML format is invalid" seems irrelevant as I got that once and it seemed to have been cured by restarting docker container.

The second one may be the problem, but what does it mean and what can I do with it?

**Comment by prononext** [comment]
I also had some hassle at the beginning to with STT and TTS. 
Be sure your User Settings in the GUI is selected correctly and maybe clear page-cache, that solved it for me. 

**librechat.yml config**

```
# Text to Speach
tts:
  elevenlabs:
    apiKey: '${TTS_API_KEY}'
    model: 'eleven_multilingual_v2'
    voices: ['your-voice-id']

# Speach to Text
stt:
  openai:
    apiKey: '${STT_API_KEY}'
    model: 'whisper-1'
```  

**LibreChat Docker ENVs:**
``` 
TTS_API_KEY=your-tts-api-key
STT_API_KEY=your-openai-api-key
```
